### PR TITLE
avoid extra copies in batchnorm inference by introducing a new op, _native_batch_norm_legit_no_training

### PR DESCRIPTION
**Pull request by bdhirsh**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #94946

### CODE REVIEW

**Review by ezyang — APPROVED** [review]
this is horrible but I am OK with getting it in for release. Is someone signed up for batch norm consolidation

**Comment by bdhirsh** [comment]
Right now nobody is signed up (cc @albanD @janeyx99 ?)

**Comment by bdhirsh** [comment]
added a test

File: torch/_decomp/decompositions.py [hunk 1]
```diff
@@ -1352,9 +1352,9 @@ def native_batch_norm_helper(
                 running_var.copy_(new_running_var)
     else:
         assert running_mean is not None and running_var is not None
-        running_mean = running_mean.to(dtype=computation_dtype, copy=True)
+        running_mean = running_mean.to(dtype=computation_dtype)
```
Comment (eellison) [inline torch/_decomp/decompositions.py:1355]: Where are the copies happening for the training version ?

File: aten/src/ATen/native/Normalization.cpp [hunk 2]
```diff
@@ -800,6 +800,11 @@ std::tuple<Tensor, Tensor, Tensor> _batch_norm_legit_no_stats_cpu(
     bool train, double momentum, double eps) {
   return batch_norm_cpu(self, weight_opt, bias_opt, Tensor(), Tensor(), train, momentum, eps);
 }
+std::tuple<Tensor, Tensor, Tensor> _batch_norm_legit_no_training(
+    const Tensor& self, const c10::optional<Tensor>& weight_opt, const c10::optional<Tensor>& bias_opt,
+    const Tensor& running_mean, const Tensor& running_var, double momentum, double eps) {
+  return batch_norm_cpu(self, weight_opt, bias_opt, const_cast<Tensor&>(running_mean), const_cast<Tensor&>(running_var), /*train=*/false, momentum, eps);
```
Comment (ngimel) [inline aten/src/ATen/native/Normalization.cpp:806]: why is this cpu only?

Comment (bdhirsh) [reply aten/src/ATen/native/Normalization.cpp:806]: Whoops 😅  will fix when i get back to my laptop.

although this code path will never actually be hit in the PT2 workflow (since the decomp for this new op is automatically opted into by inductor + any backends that run “core aten decomps”)

**Comment by bdhirsh** [comment]
The extra `.to(..., copy=True)`s will live to see another day (I had to leave them in to appease tests, although I confirmed that copies still don't show up in the batchnorm inference graph). I think they'll be easier to clean up once batch norm consolidation happens. I'm pretty sure this is failing because there's a test passing `train=False` directly to `_native_batch_norm_legit.default`, even though that op should never handle the train=False case anymore.
